plugin.js

```diff
@@ -25,9 +26,9 @@ function getName (func) {
   return func.toString().split('\n').slice(0, 2).map(s => s.trim()).join(' -- ')
 }
 
-function Plugin (parent, func, opts, isAfter, timeout) {
+function Plugin (parent, func, optsOrFunc, isAfter, timeout) {
   this.func = func
-  this.opts = opts
+  this.opts = isFunction(optsOrFunc) ? deepClone(optsOrFunc()) : optsOrFunc
```

Comment:
Why not?

```suggestion
  this.opts = optsOrFunc instanceof Function ? deepClone(optsOrFunc()) : optsOrFunc
```

Reply:
My bad, the segment copied was fixing an issue that only exists in browsers when performing cross-frame checks. I implemented your suggestion and cleaned up `isFunction`.